### PR TITLE
Rename MasterSlave module to ReadWriteSplit

### LINKED ISSUE #7307: Rename MasterSlave module to ReadWriteSplit
This is a proactive change to catch up with the movement of industry.
Primary and replica will be used as alternatives.

### PR DESCRIPTION
**Pull request by flycoolman**
Fixes #7307 .

Changes proposed in this pull request:
- Rename 'master' to 'primary'
- Rename 'Master' to 'Primary'
- Rename 'slave' to 'replica'
- Rename 'Slave' to 'Replica'
- Rename 'ms' to 'pr' if the 'ms' is the abbreviation of 'masterslave'
- Rename 'ds_m' to 'ds_p'
- Rename 'ds_ms' to 'ds_pr'
- Rename other combinations of 'master' and 'slave', such as 'master-slave', 'MasterSlave', 'MASTER', 'SLAVE', etc.

Hi @terrymanu @tristaZero ,
Please review the changes and let me know any further changes. This fix does not include the folder of 'examples'.
I have all the tests pass locally. I would like to keep this fix as small as possible, as long as all the tests can pass.  

Thanks for your patience.

### CODE REVIEW

File: shardingsphere-features/pom.xml [hunk 1]
```diff
@@ -32,7 +32,7 @@
     <modules>
         <module>shardingsphere-sharding</module>
         <module>shardingsphere-replica</module>
-        <module>shardingsphere-master-slave</module>
+        <module>shardingsphere-primary-replica</module>
```
Comment (terrymanu) [inline shardingsphere-features/pom.xml:35]: The module name should be read-write-split

File: shardingsphere-features/shardingsphere-primary-replica/shardingsphere-primary-replica-api/src/main/java/org/apache/shardingsphere/primaryreplica/api/config/PrimaryReplicaRuleConfiguration.java [hunk 2]
```diff
@@ -15,25 +15,25 @@
...
 import java.util.Map;
 
 /**
- * Master-slave rule configuration.
+ * Primary-replica rule configuration.
  */
 @RequiredArgsConstructor
 @Getter
-public final class MasterSlaveRuleConfiguration implements RuleConfiguration {
+public final class PrimaryReplicaRuleConfiguration implements RuleConfiguration {
```
Comment (terrymanu) [inline shardingsphere-features/shardingsphere-primary-replica/shardingsphere-primary-replica-api/src/main/java/org/apache/shardingsphere/primaryreplica/api/config/PrimaryReplicaRuleConfiguration.java:34]: The rule name should be ReadWriteSplitRuleConfiguration

**Review by terrymanu — CHANGES_REQUESTED** [review]

**Comment by terrymanu** [comment]
Please change the title of pr to make sense

**Comment by terrymanu** [comment]
I notice you commit `404` times which is unbelievable.
Can you reference the Contributor Guide(https://shardingsphere.apache.org/community/en/contribute/contributor/) to standardize your commit operation?

File: shardingsphere-features/pom.xml (hunk 1 above)
Comment (flycoolman) [reply shardingsphere-features/pom.xml:35]: Got it! 
If there are other places you can imagine to use other names, not just plain translate from master-slave to primary-replica, please let me know.

File: shardingsphere-features/shardingsphere-primary-replica/shardingsphere-primary-replica-api/src/main/java/org/apache/shardingsphere/primaryreplica/api/config/PrimaryReplicaRuleConfiguration.java (hunk 2 above)
Comment (flycoolman) [reply shardingsphere-features/shardingsphere-primary-replica/shardingsphere-primary-replica-api/src/main/java/org/apache/shardingsphere/primaryreplica/api/config/PrimaryReplicaRuleConfiguration.java:34]: Will change

**Comment by flycoolman** [comment]
> I notice you commit `404` times which is unbelievable.
> Can you reference the Contributor Guide(https://shardingsphere.apache.org/community/en/contribute/contributor/) to standardize your commit operation?

Sorry about this. I definitely can squash the commits.
Just FYI:
Basically the commits were based on files that were changed accordingly, as other modules have dependency on the  ReadWriteSplit module. Hopefully I did not do it out of scope.

**Comment by flycoolman** [comment]
> Please change the title of pr to make sense

Changed

**Comment by kimmking** [comment]
Hi, @flycoolman 
You should resolve the conficts first.

File: shardingsphere-features/pom.xml (hunk 1 above)
Comment (flycoolman) [reply shardingsphere-features/pom.xml:35]: > Hi, @flycoolman
> You should resolve the conficts first.
@kimmking Thanks for the suggestion. 
That will be my next action once the final decision about the module name is made.

**Comment by flycoolman** [comment]
> Hi, @flycoolman
> You should resolve the conficts first.

@kimmking Thanks for the suggestion.
That will be my next action once the final decision about the module name is made.

**Comment by terrymanu** [comment]
Maybe we need to close this pr because of the commit log is error.
It is better to open a new pr to do this.

**Comment by flycoolman** [comment]
Close this and open a new one when the changes are done.
Current status:
Still pending on the final decision about this change which will be made.

**Pull request closed without merging**